### Chapter 1: What's RepeatExplorer?
RepeatExplorer2 clustering is a computational pipeline for unsupervised identification of repeats from unassembled sequence reads. 
The pipeline uses low-pass whole genome sequence reads and performs graph-based clustering. 
Resulting clusters, representing all types of repeats, are then examined to identify and classify into repeats groups.

The analysis requires either single or paired-end reads generated by whole genome shotgun sequencing provided as a single fasta-formatted file.
Generally, paired-end reads provide significantly better results than single reads. 
Reads should be of uniform length (optimal size range is 100-200 nt) 
and the number of analyzed reads should represent less than 1x genome equivalent (genome coverage of 0.01 - 0.50 x is recommended). 
Reads should be quality-filtered (recommended filtering : quality score >=10 over 95% of bases and no Ns allowed)
and only complete read pairs should be submitted for analysis. 
When paired reads are used, input data must be interlaced format as fasta file

To do:
find a way to compare RN and RepeatExplorer given 1 only allows raw reads and another only assembled reads
One way: 
1) run assembly on assembled reads
2) run TR and RM on assembled, and RE on unassembled (note - QC needed, and paired end reads are optimal)


Optimal workflow (example data from practical http://repeatexplorer.org/repeatexplorer/wp-content/Manual_for_practical_training_2023.pdf):
1) Use Trimmomatic program to remove low quality reads and trim read ends. 
```bash
cp /usr/share/trimmomatic/*.fa .
# Remove first 10 nt, min length must be 90
TrimmomaticPE -phred33 SRR089356_1.fastq.gz SRR089356_2.fastq.gz \
 SRR089356_1_clean.fastq.gz SRR089356_1_unpaired.fastq.gz \
 SRR089356_2_clean.fastq.gz SRR089356_2_unpaired.fastq.gz \
 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15 SLIDINGWINDOW:4:10 CROP:100 HEADCROP:10
MINLEN:90
# Check statistics of fastq files:
seqkit stats *fastq.gz
# Run fastqc on clean data:
fastqc *clean*.fastq.gz
```

2) Sample to required coverage:
```bash
# Paired end read sampling:
seqtk sample -s 10 SRR089356_1_clean.fastq.gz 5000 >
SRR089356_1_clean_sample.fastq
seqtk sample -s 10 SRR089356_2_clean.fastq.gz 5000 >
SRR089356_2_clean_sample.fastq
```

3) Interleaved pairs into single file:
```bash
# Make interleaved FASTQ
seqtk mergepe SRR089356_1_clean_sample.fastq SRR089356_2_clean_sample.fastq
> SRR089356_clean_sample_merged.fastq
# Convert to FASTA
seqtk seq -A SRR089356_clean_sample_merged.fastq >
SRR089356_clean_sample_merged.fasta
```

4) Run RepeatExplorer with default settings:
```bash
# Run clustering with default settings
cd ~/repeatexplorer
singularity exec -e --bind ${PWD}:/data/ repex_tarean seqclust -p -v
/data/re_output_run1 /data/single_species/SRR089356_clean_sample_merged.fasta
```

---
### Chapter 2: testing on SRA data (random)
First run on raw data downloaded from SRA (no QC, etc.)
```bash
prefetch SRX29907886
fastq-dump --outdir ./ --gzip --skip-technical --readids --read-filter pass --dumpbase --split-3 SRR34774499.sra
seqtk seq -a SRR34774499_pass.fastq.gz > SRR34774499.fasta
singularity exec --bind ${PWD}:/data/ repex_tarean.sif seqclust -v /data/re_output /data/SRR34774499.fasta
```

Time:
2025-08-01 13:37:11,122 - __main__ - INFO -
2025-08-01 18:55:14,398 - __main__ - INFO -
10 mb file - 5.5 hours

RepearExplorer instruction states:
"The analysis requires either single or paired-end reads generated by whole genome shotgun sequencing provided as a single fasta-formatted file. Generally, paired-end reads provide significantly better results than single reads. Reads should be of uniform length (optimal size range is 100-200 nt) and the number of analyzed reads should represent less than 1x genome equivalent (genome coverage of 0.01 - 0.50 x is recommended). Reads should be quality-filtered (recommended filtering : quality score >=10 over 95% of bases and no Ns allowed) and only complete read pairs should be submitted for analysis. When paired reads are used, input data must be interlaced format as fasta file"

In this case I am using single reads, 20000 reads ~250 nt long each which is suboptimal

Now need to:
1) understand output
2) understand metrics to calculate coverage and accuracy
3) understand software to calculate efficiency (if it's necesary)
4) choose software to do assembly to compare to RN software
5) calculate coverage and accuracy

Notes:
- has a lot of dependencies, so best run with Singularity container. Makes installation process harder

-----
Citation:
Petr Novák, Pavel Neumann, Jiří Pech, Jaroslav Steinhaisl, Jiří Macas, RepeatExplorer: a Galaxy-based web server for genome-wide characterization of eukaryotic repetitive elements from next-generation sequence reads, Bioinformatics, Volume 29, Issue 6, March 2013, Pages 792–793, https://doi.org/10.1093/bioinformatics/btt054
-----

---
### Chapter 2: Drosophila Melanogaster A4:
1) using downloaded fastq paired end reads (fastg.gz) trim adapters and make length 120 bp:

```bash
# I downloaded Illumina adapter sequences with trimmomatic 0.39. I put it into raw_QC folder and copied fastq.gz files in there as well. Since I used conda installation of trimmomatic it used adapters that come with the software from conda and downloading adapters was not necesary
trimmomatic PE -threads 8 -phred33 \
  SRR29479670_1.fastq.gz SRR29479670_2.fastq.gz \
  R1_clean.fastq.gz R1_unpaired.fastq.gz \
  R2_clean.fastq.gz R2_unpaired.fastq.gz \
  ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:True \
  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:20 CROP:120 MINLEN:120
```

```output
ILLUMINACLIP: Using adapter file from Trimmomatic installation folder: /home/aygera/miniconda3/envs/bioinfo/share/trimmomatic-0.40-0/adapters/TruSeq3-PE.fa
Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Input Read Pairs: 87332806 Both Surviving: 77255629 (88.46%) Forward Only Surviving: 4906155 (5.62%) Reverse Only Surviving: 2485299 (2.85%) Dropped: 2685723 (3.08%)
TrimmomaticPE: Completed successfully
```

# Check statistics of fastq files:
seqkit stats *fastq.gz
# Run fastqc on clean data:
fastqc *clean*.fastq.gz

```output
processed files:  6 / 6 [======================================] ETA: 0s. done
file                    format  type    num_seqs         sum_len  min_len  avg_len  max_len
R1_clean.fastq.gz       FASTQ   DNA   77,255,629   9,270,675,480      120      120      120
R1_unpaired.fastq.gz    FASTQ   DNA    4,906,155     588,738,600      120      120      120
R2_clean.fastq.gz       FASTQ   DNA   77,255,629   9,270,675,480      120      120      120
R2_unpaired.fastq.gz    FASTQ   DNA    2,485,299     298,235,880      120      120      120
SRR29479670_1.fastq.gz  FASTQ   DNA   87,332,806  13,099,920,900      150      150      150
SRR29479670_2.fastq.gz  FASTQ   DNA   87,332,806  13,099,920,900      150      150      150
```

2) Sub-sample to required coverage:
```bash
# Paired end read sampling:
seqtk sample -s 10 R1_clean.fastq.gz 300000 > R1_clean_sample.fastq
seqtk sample -s 10 R2_clean.fastq.gz 300000 > R2_clean_sample.fastq
```

3) interleave paired end reads and convert to fasta
```bash
seqtk mergepe R1_clean_sample.fastq R2_clean_sample.fastq > merged.fastq
seqtk seq -A merged.fastq > merged.fasta
```

4) run RepeatExplorer
```bash
singularity exec -e --bind ${PWD}:/data/ repex_tarean.sif seqclust -p -v /data/re_output /data/merged.fasta -c 24
```

This happened to finish in less than hour. Started at 17:45 and finished at 18:33 = 48 mins. But I accidentally used 250.000 reads instead of 300,000. So I might want to rerun it.

Masking sequences with it:
```bash
cat contigs.fasta TAREAN_consensus_rank_*.fasta > custom_repeats_raw.fa
../../tools/RepeatMasker/RepeatMasker/RepeatMasker  -lib ./custom_repeats_raw.fa -xsmall -gff ./a4_assembly.fna -pa 24
```
